src/vfs2.c

```diff
@@ -970,22 +975,42 @@ static struct wal_index_full_hdr initial_full_hdr(struct wal_hdr whdr)
 	return ihdr;
 }
 
-static void set_mx_frame(struct wal_index_full_hdr *ihdr,
+static void pgno_ht_insert(uint16_t *ht, size_t len, uint16_t fx, uint32_t pgno)
```

Comment:
The "maximum half full" rule is enforced by storing hashes for only the first 4062 WAL frames in the first hash table, which has space for 8192 hashes. For subsequent hash tables it's the same story but with 4096 page numbers. So assuming we support long WALs with more than 4062 frames, we would implement this at a higher level, by calling `pgno_ht_insert` on one of the subsequent tables instead of the first one.

Reply:
Gotcha, I thought this one was the higher level API. But is the implementation with several hash tables coming in a later PR or how do we plan to do that?